internal/pubsub/mempubsub/mem.go

```diff
@@ -91,7 +70,15 @@ func (t *topic) SendBatch(ctx context.Context, ms []*driver.Message) error {
 func (*topic) IsRetryable(error) bool { return false }
 
 // As implements driver.Topic.As.
-func (t *topic) As(i interface{}) bool { return false }
+// It supports *topic for NewSubscription (see below).
```

Comment:
Hmm, I don't think this is useful to expose externally for `As`, because the `topic` type isn't exported. I suggest leaving `As` returning `false` and doing a regular type cast to `*topic` in `NewSubscription`.

Reply:
What would you like me to cast to `*topic`? Not `*pubsub.Topic`—that is a concrete type.

This is approximately Ross's second design in #970, the one you preferred. In Ross's words,

> One could also get fancy with As() methods and instead make NewSubscription just take in a `*pubsub.Topic`

Can you describe in detail the design you actually want?